### PR TITLE
Optimize bond accrual history calculation #740

### LINKED ISSUE #740: Optimize bond accrual history calculation
Split from #739 (O1).

## Problem
Bond historical valuation repeatedly rebuilds accrual from the entry posting date for each output day. Entry selection also rescans account entries for each date/instrument, which can make long histories grow roughly quadratically.

## Goal
Calculate bond accrual history once per effective entry/instrument interval and reuse it while advancing through dates.

## Suggested approach
- Partition bond entries by instrument and sort once.
- Advance an entry cursor as valuation dates progress.
- Compute each effective entry's accrual series once, or maintain incremental accrual state that emits daily values.
- Reuse the computed series in chart and net-worth composition.
- Keep financial calculation rules in Domain and orchestration in Application.

## Acceptance / validation
- Output remains equivalent to the current implementation across multi-year histories, intraday entries, opening boundaries, zero holdings, rate changes, missing calculation methods, and capitalization boundaries.
- Preserve current rounding and 365-day capitalization semantics.
- Add performance coverage for 1/5/10-year histories and compare CPU/allocation behavior.

## Evidence from audit
`BondAccountEntry.GetPriceAt`, `BondAccount.GetDailyPrice`, `BondEntryExtension`, and `NetWorthService.GetNetWorth` were identified as the relevant paths.

### PR DESCRIPTION
**Pull request by avresial**
Closes #740

## Summary
- Replace per-day bond entry rescans and repeated full-history accrual rebuilds with grouped, sorted entry intervals and bounded accrual series.
- Preserve same-day precedence, carry-in entries, rate changes, capitalization, rounding, zero omission, and output order.
- Add reference-equivalence and 1/5/10-year performance coverage.

## Validation
- 37 focused bond tests passed
- 1,026 unit tests passed
- 326 integration tests passed
- 9 architecture tests passed
- Solution build: 0 warnings, 0 errors
- Scoped format verification and git diff checks passed

Pure refactor with behavior-equivalence coverage; no changelog entry required.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved daily bond price accuracy across multiple instruments, posting dates, carried-in entries, rate changes, and leap-year boundaries.
  - Preserved correct handling of intraday precedence, net-zero balances, missing bond details, and invalid date ranges.

- **Performance**
  - Improved efficiency for long-term daily price histories, including multi-year calculations.

- **Tests**
  - Expanded coverage for accrual accuracy, boundary conditions, capitalization, precision, and long-horizon performance.

### CODE REVIEW

**Comment by avresial** [comment]
Remediation complete for PR #756 at head `a37a73e96657d20cbc05eb3a21bdd6c5e6a5dfab`.

Outcome:
- The original run [34566615051](https://github.com/avresial/FinanceManager/actions/runs/34566615051) failed only `NbpResilienceRegistrationTests.NbpRegistration_DoesNotDoubleRetry_AttemptsAreBounded` (expected 2–4 attempts, observed 1). That test is outside PR #756’s changed paths; the exact-head test class passed 3/3 and the full integration project passed 326/326 locally.
- Pushed focused commit [a37a73e](https://github.com/avresial/FinanceManager/commit/a37a73e96657d20cbc05eb3a21bdd6c5e6a5dfab), replacing the tautological performance assertions with explicit per-horizon CPU/allocation budgets and applying collection expressions.
- Local validation passed: focused performance tests 10/10, full unit tests 1,026/1,026, full integration tests 326/326, format verification, and strict solution build with 0 warnings/0 errors.
- Replacement run [34589138471](https://github.com/avresial/FinanceManager/actions/runs/34589138471) is successful for build, code quality, integration tests, security scan, architecture tests, unit tests, and CodeQL; publish/deploy jobs were skipped by workflow policy. CodeRabbit also passed with review completed and no actionable comments.
- Both known inline review threads are resolved and outdated; the performance thread was answered with the exact budgets and the style thread was confirmed addressed by CodeRabbit.
- PR remains open, mergeable, and `CLEAN`. Its temporary `in progress` label is cleared. No re-review request was needed because reviews are `COMMENTED`, not `CHANGES_REQUESTED`.
- Issue #740 remains open with `ready for merge`, `optimization`, `backend`, and `size/M`; `in progress` was removed after checks passed and no competing PR #756 agent remained active.
- No merge, close, review dismissal, force-push, or unrelated-work modification was performed.

File: code/FinanceManager.Tests.Unit/Domain/Entities/Bonds/BondAccountAccrualPerformanceTests.cs [hunk 1]
```diff
@@ -0,0 +1,637 @@
...
+        var stopwatch = Stopwatch.StartNew();
+        var dailyPrices = account.GetDailyPrice(startDate, endDate, detailsList);
+        stopwatch.Stop();
+        var allocatedBytes = GC.GetAllocatedBytesForCurrentThread() - startAllocated;
+
+        output.WriteLine($"[1-Year History] Elapsed: {stopwatch.ElapsedMilliseconds} ms, Allocated: {allocatedBytes:N0} bytes, Days: {dailyPrices.Count}");
+
+        // Assert: Performance bounds (non-flaky, sanity check only)
+        Assert.True(stopwatch.ElapsedMilliseconds >= 0);
+        Assert.True(allocatedBytes >= 0);
```
[the remark this answers is not in the document]
Comment (avresial) [reply code/FinanceManager.Tests.Unit/Domain/Entities/Bonds/BondAccountAccrualPerformanceTests.cs]: Addressed in commit `a37a73e`. The tautological checks are replaced with explicit upper bounds for the optimized calculation at each horizon: 1 year ≤1,000 ms and 5 MB, 5 years ≤2,500 ms and 15 MB, and 10 years ≤5,000 ms and 30 MB. The focused performance tests pass 10/10, and the full unit suite passes 1,026/1,026. The thread is outdated because these assertions are now on the current head; replacement CI is running.
